djstripe/models.py

```diff
@@ -497,10 +497,11 @@ def sync_current_subscription(self, cu=None):
             return sub_obj
 
     def update_plan_quantity(self, quantity, charge_immediately=False):
+        sub = self.stripe_customer.subscription
+        if not sub:
+            raise SubscriptionUpdateFailure("Customer does not have a subscription with Stripe")
```

Comment:
The problem is that, as currently written, sync_current_subscription won't actually do anything in this case, although maybe it should.